higher-order-functions/eloquent/exercise.js

```diff
@@ -1,11 +1,25 @@
-// Challenge 1. Flattening
+// Challenge 1. Flattening 
 let arrays = [[1, 2, 3], [4, 5], [6]];
+function flatten(arrays) {
+  return arrays.reduce((accumelator,currentValue) => {
+  return accumelator.concat(currentValue)
+})
+}
 
 // Your code here.
 // → [1, 2, 3, 4, 5, 6]
 
 // Challenge 2. Your own loop
 // Your code here.
+function loop(n,condition,update,body) {
+  for(let n = n; condition; update)
```

Comment:
You cant have a function reference as the `test condition` in for loop because test condition only understand boolean. This will go in infinite loop.